GitCommands/Settings/Setting.cs

```diff
@@ -173,13 +174,37 @@ private object GetValue(string name)
                 return SettingsSource
                     .GetValue<object>(name, null, value =>
                     {
-                        switch (Type.GetTypeCode(typeof(T)))
+                        var type = typeof(T);
+                        var underlyingType = Nullable.GetUnderlyingType(type) ?? type;
+
+                        switch (Type.GetTypeCode(underlyingType))
                         {
                             case TypeCode.String:
-                                return value;
+                                return (string)value;
+                            case TypeCode.Object:
+                                try
+                                {
+                                    return JsonConvert
+                                        .DeserializeObject<T>(value);
```

Comment:
I must have totally zoned out when reviewed your previous PR, what do we serialize/deserialize to JSON? All the settings files I've seen were all XML and XML within XML...

Reply:
If value is object, now it is a structure, besides standard types.
This does not affect the xml structure.

Reply:
This also allows you to collect settings into one logical group.

Reply:
I'm trying to establish - is it a new or an existing functionality?

Reply:
Fixed convertation for simple types. Relatively new.

Reply:
For Int, Byte, Float, Enum... changed to use type converter